### PR TITLE
use edf wrr to cover the original smooth wrr

### PR DESCRIPTION
**Pull request by CodingSinger**
the time complexity of edf schedule wrr is O (logn), better than O (n) of smoothWeightedRR LB.
Relate #1161

### CODE REVIEW

File: pkg/upstream/cluster/loadbalancer.go [hunk 1]
```diff
@@ -152,80 +152,49 @@ func (lb *roundRobinLoadBalancer) HostNum(metadata api.MetadataMatchCriteria) in
...
 }
 
+func (lb *WeightedRRLoadBalancer) hostWeight(item WeightItem) float64 {
+	host := item.(types.Host)
+	return float64(host.Weight())
+}
+// do unweighted (fast) selection
+func (lb *WeightedRRLoadBalancer) unweightChooseHost(context types.LoadBalancerContext) types.Host {
+	targets := lb.hosts.Hosts()
+	total := len(targets)
```
Comment (wangfakang) [inline pkg/upstream/cluster/loadbalancer.go:193]: need check value of `total`, if it's 0 should return nil.

File: pkg/upstream/cluster/loadbalancer.go [hunk 2]
```diff
@@ -46,7 +46,7 @@ func init() {
 	}
 	RegisterLBType(types.RoundRobin, rrFactory.newRoundRobinLoadBalancer)
 	RegisterLBType(types.Random, newRandomLoadBalancer)
-	RegisterLBType(types.WeightedRoundRobin, newSmoothWeightedRRLoadBalancer)
+	RegisterLBType(types.WeightedRoundRobin, newWeightedRRLoadBalancer)
```
Comment (wangfakang) [inline pkg/upstream/cluster/loadbalancer.go:49]: Maybe better that  named `newWeightedWRRLoadBalancer` .

File: pkg/upstream/cluster/loadbalancer.go (hunk 1 above)
Comment (CodingSinger) [reply pkg/upstream/cluster/loadbalancer.go:193]: Already judged in the `EdfLoadbalancer#ChooseHost`.

File: pkg/upstream/cluster/loadbalancer.go (hunk 2 above)
Comment (CodingSinger) [reply pkg/upstream/cluster/loadbalancer.go:49]: I think WeigthtedRR = WRR?

Comment (wangfakang) [reply pkg/upstream/cluster/loadbalancer.go]: ok

**Pull request merged**